src/reducer.js

```diff
@@ -68,8 +69,10 @@ function immutableOnce (object) {
  * @returns {Object} a value cleaned of any `null`s
  */
 function recursiveClean (value, model) {
-  let output = Array.isArray(value) ? [] : {}
-  _.forEach(value, (subValue, key) => {
+  let isValueArray = Array.isArray(value)
+  let output = isValueArray ? [] : {}
+  let iteratorFn = isValueArray ? _.forEach : _.forIn
+  iteratorFn(value, (subValue, key) => {
```

Comment:
Here's the actual fix (using `_.forIn` when an object and `_.forEach` when an array.